src/rpp/rpp/operators/combine_latest.hpp

```diff
@@ -0,0 +1,173 @@
+            }, m_values);
+        }
+    };
+
+    /**
+     * \brief Templated helper function for subscribing to variadic 'other' observables.
+     *
+     * \param observables is a variadic tuple. The first observable is the current observable and rest are the other observables.
+     * \param subscriber is the downstream subscriber.
+     * \param coordinator manages the cache of emission from the observables and coordinate dispatching.
+     */
+    template<size_t...I>
+    static void subscribe_other_observables(std::index_sequence<I...>, // Used in compile time for variadic expansion
+                                            const auto& observables,
+                                            const auto& subscriber,
+                                            const auto& coordinator)
+    {
+        // +1 because the first element in tuple is the current observable, and you want to subscribe to the 'other' observables.
+        // (Use variadic expansion to iterate the observables)
+        (subscribe_observable<I + 1>(std::get<I>(observables), subscriber, coordinator), ...);
+    }
+
+    template<size_t I, constraint::observable TObs>
+    static void subscribe_observable(const TObs& observable, const auto& subscriber, const auto& coordinator)
+    {
+        using ValueType = utils::extract_observable_type_t<TObs>;
+        observable.subscribe(create_subscriber_with_state<ValueType>(
+            subscriber.get_subscription().make_child(),
+            subscriber,
+            [coordinator](auto&& value, const auto& subscriber)
+            {
+                coordinator->template offer_next_value<I>(std::forward<decltype(value)>(value), subscriber);
+            },
+            utils::forwarding_on_error{},
+            [coordinator](const auto& subscriber)
+            {
+                coordinator->offer_completed(subscriber);
+            }));
+    }
+
+public:
+    template<constraint::subscriber_of_type<DownstreamType> TSub>
+    auto operator()(TSub&& subscriber) const
+    {
+        // TODO: Avoid copying the combiner?
+        auto coordinator = std::make_shared<combine_latest_coordinator_t<Type, utils::extract_observable_type_t<TObservable>...>>(m_combiner);
+
+        // Subscribe to other observables and redirect on_next event to coordinator
+        subscribe_other_observables(std::index_sequence_for<TObservable...>{}, m_observables, subscriber, coordinator);
+
+        // Redirect values from this observable to the coordinator for value composition
+        auto on_next = [coordinator](auto&& value, const auto& subscriber)
+        {
+            coordinator->template offer_next_value<INDEX_OF_SOURCE_TYPE>(std::forward<decltype(value)>(value), subscriber);
+        };
+        auto on_completed = [coordinator](const auto& subscriber)
+        {
+            coordinator->offer_completed(subscriber);
+        };
+
+        return create_subscriber_with_state<Type>(std::forward<TSub>(subscriber),
```

Comment:
```suggestion
        auto sub = subscriber.get_subscription().make_child(),
        return create_subscriber_with_state<Type>(std::move(sub),
                                                  std::forward<TSub>(subscriber),
```

Reply:
In what cases we need to create a children subscription?

Reply:
Discussed on gitter